### PR TITLE
Correcting Misleading Test Skip Reasons.

### PR DESCRIPTION
**Pull request by drivanov**
## Description

This PR is similar to [PR#7749](https://github.com/dmlc/dgl/pull/7749) and it addresses the following warnings
```
SKIPPED [60] ../../../usr/local/lib/python3.10/dist-packages/_pytest/unittest.py:357: GPUCachedFeature requires a Volta or later generation NVIDIA GPU.
```
These warnings occur when certain GraphBolt tests are run on a CPU (`F._default_context_str != "GPU"`) on a machine with a GPU with  `device_capability=8`.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by Rhett-Ying** [comment]
@drivanov pls fix the lint check failure first.

File: tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py [hunk 1]
```diff
@@ -16,11 +16,21 @@ def to_on_disk_numpy(test_dir, name, t):
     np.save(path, t.cpu().numpy())
     return path
 
+def _skip_condition_CachedFeature():
+    return (F._default_context_str != "gpu") or \
+        (torch.cuda.get_device_capability()[0] < 7)
+
+
+def _reson_to_skip_CachedFeature():
```
Comment (mfbalin) [inline tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py:24]:
```suggestion
def _reason_to_skip_cached_feature():
```

File: tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py [hunk 2]
```diff
@@ -16,11 +16,21 @@ def to_on_disk_numpy(test_dir, name, t):
...
+    if F._default_context_str != "gpu":
+        return "GPUCachedFeature tests are available only on GPU."
```
Comment (mfbalin) [inline tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py:26]:
```suggestion
        return "GPUCachedFeature tests are available only when testing the GPU backend."
```

File: tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py [hunk 3]
```diff
@@ -17,10 +17,22 @@ def to_on_disk_numpy(test_dir, name, t):
     return path
 
 
+def _skip_condition_CachedFeature():
```
Comment (mfbalin) [inline tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py:20]:
```suggestion
def _skip_condition_cached_feature():
```

File: tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py [hunk 4]
```diff
@@ -17,10 +17,22 @@ def to_on_disk_numpy(test_dir, name, t):
...
+    return (F._default_context_str != "gpu") or (
+        torch.cuda.get_device_capability()[0] < 7
+    )
+
+
+def _reason_to_skip_CachedFeature():
```
Comment (mfbalin) [inline tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py:26]: same here.

Comment (drivanov) [reply tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py]: DONE

File: tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py (hunk 3 above)
Comment (drivanov) [reply tests/python/pytorch/graphbolt/impl/test_gpu_cached_feature.py]: DONE

**Comment by Rhett-Ying** [comment]
@dgl-bot

**Comment by Rhett-Ying** [comment]
@mfbalin could you please take another round of review?

**Review by mfbalin — APPROVED** [review]

**Comment by Rhett-Ying** [comment]
@dgl-bot

**Pull request merged**